src/azure-cli/azure/cli/command_modules/network/custom.py

```diff
@@ -2895,14 +2895,100 @@ def set_lb_frontend_ip_configuration(
     return parent
 
 
-def create_lb_backend_address_pool(cmd, resource_group_name, load_balancer_name, item_name):
-    BackendAddressPool = cmd.get_models('BackendAddressPool')
+def create_lb_backend_address_pool(cmd, resource_group_name, load_balancer_name, backend_address_pool_name,
+                                   backend_addresses=None, backend_addresses_config_file=None):
+    def _process_vnet_name_and_id(vnet):
+        if vnet and not is_valid_resource_id(vnet):
+            vnet = resource_id(
+                subscription=get_subscription_id(cmd.cli_ctx),
+                resource_group=resource_group_name,
+                namespace='Microsoft.Network',
+                type='virtualNetworks',
+                name=vnet)
+        return vnet
+    if backend_addresses_config_file:
+        if not isinstance(backend_addresses_config_file, list):
+            raise CLIError('Config file must be a list. Please see example as a reference.')
+        for addr in backend_addresses_config_file:
+            if not isinstance(addr, dict):
+                raise CLIError('Each address in config file must be a dictionary. Please see example as a reference.')
     ncf = network_client_factory(cmd.cli_ctx)
     lb = ncf.load_balancers.get(resource_group_name, load_balancer_name)
-    new_pool = BackendAddressPool(name=item_name)
-    upsert_to_collection(lb, 'backend_address_pools', new_pool, 'name')
-    poller = ncf.load_balancers.create_or_update(resource_group_name, load_balancer_name, lb)
-    return get_property(poller.result().backend_address_pools, item_name)
+    (BackendAddressPool,
+     LoadBalancerBackendAddress,
+     VirtualNetwork) = cmd.get_models('BackendAddressPool',
+                                      'LoadBalancerBackendAddress',
+                                      'VirtualNetwork')
+    # Hope we can remove this logic
+    if cmd.supported_api_version(max_api='2020-03-01') or lb.sku.name.lower() == 'basic':
+        new_pool = BackendAddressPool(name=backend_address_pool_name)
+        upsert_to_collection(lb, 'backend_address_pools', new_pool, 'name')
+        poller = ncf.load_balancers.create_or_update(resource_group_name, load_balancer_name, lb)
+        return get_property(poller.result().backend_address_pools, backend_address_pool_name)
+
+    addresses_pool = []
+    addresses_pool.extend(backend_addresses)
+    addresses_pool.extend(backend_addresses_config_file)
```

Comment:
>backend_addresses_config_file [](start = 26, length = 29)

Better to tell users only one of backend_addresses and backend_addresses_config_file can be used at the same time and also add verification. Otherwise users are confused about the behavior if they provide both.